tests/arrays/test_arrow_string_array.py

```diff
@@ -0,0 +1,89 @@
+from tests.dtypes import PYARROW_STRING_ARGS
+from tests.utils import powerset
+
+
+@pytest.mark.parametrize("data", powerset(["pd", np.str_("pd")]))
+@pytest.mark.parametrize(("dtype", "target_dtype"), PYARROW_STRING_ARGS.items())
+def test_construction_dtype(
+    data: tuple[str | np.str_, ...], dtype: PyArrowStrDtypeArg, target_dtype: type
+) -> None:
+    dtype_notna = target_dtype if data else None
+    check(pd.array([*data], dtype=dtype), ArrowStringArray, dtype_notna)
+    check(pd.array([*data, *data], dtype=dtype), ArrowStringArray, dtype_notna)
+
+    dtype_na = target_dtype if data else NAType
+    check(pd.array([*data, np.nan], dtype=dtype), ArrowStringArray, dtype_na)
+    check(pd.array([*data, *data, np.nan], dtype=dtype), ArrowStringArray, dtype_na)
+
+    if TYPE_CHECKING:
+        assert_type(pd.array([], dtype=pd.StringDtype("pyarrow")), ArrowStringArray)
+        assert_type(pd.array([], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(
+            pd.array([np.nan], dtype=pd.StringDtype("pyarrow")), ArrowStringArray
+        )
+        assert_type(pd.array([np.nan], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(pd.array(["1"], dtype=pd.StringDtype("pyarrow")), ArrowStringArray)
+        assert_type(pd.array(["1"], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(
+            pd.array(["1", "2"], dtype=pd.StringDtype("pyarrow")), ArrowStringArray
+        )
+        assert_type(pd.array(["1", "2"], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(
+            pd.array(["1", np.nan], dtype=pd.StringDtype("pyarrow")), ArrowStringArray
+        )
+        assert_type(pd.array(["1", np.nan], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(
+            pd.array([np.str_("1")], dtype=pd.StringDtype("pyarrow")), ArrowStringArray
+        )
+        assert_type(pd.array([np.str_("1")], dtype="string[pyarrow]"), ArrowStringArray)
+
+        assert_type(
+            pd.array([np.str_("1"), np.str_("2")], dtype=pd.StringDtype("pyarrow")),
+            ArrowStringArray,
+        )
+        assert_type(
+            pd.array([np.str_("1"), np.str_("2")], dtype="string[pyarrow]"),
+            ArrowStringArray,
+        )
+
+        assert_type(
+            pd.array([np.str_("1"), np.nan], dtype=pd.StringDtype("pyarrow")),
+            ArrowStringArray,
+        )
+        assert_type(
+            pd.array([np.str_("1"), np.nan], dtype="string[pyarrow]"), ArrowStringArray
+        )
+
+        assert_type(
+            pd.array(["1", np.str_("2")], dtype=pd.StringDtype("pyarrow")),
+            ArrowStringArray,
+        )
+        assert_type(
+            pd.array([np.str_("1"), "2"], dtype="string[pyarrow]"), ArrowStringArray
+        )
+
+
+def test_dtype() -> None:
+    arr = pd.array(["a"], "string[pyarrow]")
+    check(assert_type(arr.dtype, "pd.StringDtype[Literal['pyarrow']]"), pd.StringDtype)
```

Comment:
I'm not a fan of `pd.StringDtype[Literal["pyarrow"]]` as a type, but I don't see an easy way around it without introducing stub-specific classes.  It seems a Generic can't be based on a choice of literal string constants.